src/js/form-builder.js

```diff
@@ -1024,6 +1031,44 @@
       let field = utils.markup('li', optionInputs);
 
       return field.outerHTML;
+      };
+
+    var cloneItem = function cloneItem(currentItem) {
+        var currentId = currentItem.attr("id");
+        var cloneName = currentItem.attr("type");
+        var ts = new Date().getTime();
+        cloneName = cloneName + "-" + ts;
+
+        var $clone = currentItem.clone();
+
+        $clone.find('[id]').each(function () { this.id = this.id.replace(currentId, lastID) });
+
+        $clone.find('[for]').each(function () { this.setAttribute('for', this.getAttribute('for').replace(currentId, lastID)) });
+
+        $clone.each(function (i, e) {
+            $("e:not(.form-elements)").each(function () {
+                var newName = this.getAttribute('name');
+                newName = newName.substring(0, (newName.lastIndexOf('-') + 1));
+                newName = newName + ts.toString();
+                this.setAttribute('name', newName);
+            });
+
+        });
+
+        $clone.find(".form-elements").find(":input").each(function () {
+            if (this.getAttribute('name') == 'name') {
+                var newVal = this.getAttribute('value');
+                newVal = newVal.substring(0, (newVal.lastIndexOf('-') + 1));
+                newVal = newVal + ts.toString();
+                this.setAttribute('value', newVal);
+            }
+        });
+
+        $clone.attr("id", lastID);
+        $clone.attr("name", cloneName);
+        $clone.addClass("cloned");

```

Comment:
Cloned fields with options like select and checkbox groups will need to have their options made sortable again. Adding `$('.sortable-options', $clone).sortable();` below this line seems to fix that.